### DISCUSSION TITLE
Trying to calculate Berry Phase 1D spin analogue of SSH model

### DISCUSSION THREAD

**Discussion by VivekSah2001** [QuSpin/QuSpin#613] [Q&A]
I am trying to write the code for Berry Phases calculation for 1D spin analogue SSH Model under open boundary condition (OBC) and Periodic boundary condition (PBC) by perturbing a bond J1 on the system, and I am using 1d spin basis with Nup = L//2 `spin_basis_1d((L),pauli=False,Nup=L//2)` . For some reason, I am getting reverse behavior in PBC and OBC. For clarity, analytically, Berry Phase in OBC should transition from 0 to pi at J1/J2 =1 in OBC and pi in PBC for all J1/J2, but **I am getting reverse behavior, pi for OBC, and that transition from 0 to pi in PBC**. I cannot understand the issue as I have correctly considered all the couplings in the site coupling list properly.

**Comment by mzubair199** [comment]
Can you share your code? Then I can suggest something!
